test/functional/test_framework/mininode.py

```diff
@@ -1792,7 +1793,14 @@ def send_message(self, message, pushbuf=False):
             tmsg += h[:4]
         tmsg += data
         with mininode_lock:
-            self.sendbuf += tmsg
+            if (len(self.sendbuf) == 0 and not pushbuf):
```

Comment:
nit: `if not self.sendbuf and not pushbuf:`. From PEP8: "For sequences, (strings, lists, tuples), use the fact that empty sequences are false."

Reply:
No need to fix this "style nit". `len(some_list)==0` is stricter than just casting any object/primitive type to boolean. Also it is more clear to read, imo.

Reply:
@MarcoFalke I believe that for values of the `list` type `if len(some_list)==0` is actually identical to `if not some_list`, and in general I'd argue for following things like PEP8.

Reply:
ACK with or without this nit change. I have a slight preference for being pythonic and following PEP8. Leaving the longer form (`if (len(self.sendbuf) ==0` would catch any bugs where someone set `self.sendbuf` to something other than a list, but I don't think that's a good justification.

Reply:
Either way I think the brackets should at least be removed

Reply:
@sipa, it is identical. Though, in the case that `some_list` is `False` or `None` or `0` or anything else that casts to `False`, `len(some_list)` will fail appropriately. I don't think it is a downside to fail in the test framework when a parameter has the wrong type.

Reply:
I think to be hyper pedantic, `if len(self.sendbuf) == 0` is equivalent to:

```python
if not hasattr(self.sendbuf, '__len__'):
    raise TypeError
else if not self.sendbuf:
    # do stuff
```

ie, it basically has an implicit assert that the type of `self.sendbuf` has a length property.

In general we don't litter the Python code with type checks, even though it's not a type safe language and someone could call into functions with the wrong typed parameters.

Reply:
👍 on @jimpo and @MeshCollider suggestions.